Translate "Writing markup with jsx" page (#424)
reactjs/it.react.dev: src/content/learn/writing-markup-with-jsx.md

</DiagramGroup>

Ma mentre il Web diventava sempre più interattivo, la logica determinava sempre più il contenuto. JavaScript era responsabile dell'HTML! Ecco perché **in React, la logica di rendering e il markup vivono insieme nello stesso posto, ovvero nei componenti.**

<DiagramGroup>

<Diagram name="writing_jsx_sidebar" height={330} width={325} alt="React component with HTML and JavaScript from previous examples mixed. Function name is Sidebar which calls the function isLoggedIn, highlighted in yellow. Nested inside the function highlighted in purple is the p tag from before, and a Form tag referencing the component shown in the next diagram.">

> `Sidebar.js`componente React

Reviewer: ```suggestion
Componente React `Sidebar.js`
```
Credo sia più chiaro tradurre la didascalia in quest'ordine. Cosa ne pensi? :)

</Diagram>

<Diagram name="writing_jsx_form" height={330} width={325} alt="React component with HTML and JavaScript from previous examples mixed. Function name is Form containing two handlers onClick and onSubmit highlighted in yellow. Following the handlers is HTML highlighted in purple. The HTML contains a form element with a nested input element, each with an onClick prop.">

> `Form.js` componente React

Reviewer: ```suggestion
Componente React `Form.js`
```
Come sopra 😄

</Diagram>

</DiagramGroup>

> Mantenere la logica di rendering e il markup di un pulsante insieme garantisce che rimangano sincronizzati ogni volta che vengono modificati. Al contrario, i dettagli non correlati, come il markup di un pulsante e il markup di una barra laterale, sono isolati l'uno dall'altro, rendendolo più sicuro modificare ognuno di essi da solo.

Reviewer: ```suggestion
Mantenere la logica di rendering e il markup di un pulsante insieme garantisce che rimangano sincronizzati ogni volta che vengono modificati. Al contrario, i dettagli non correlati, come il markup di un pulsante e il markup di una barra laterale, sono isolati l'uno dall'altro, rendendo più sicuro modificare ognuno di essi da solo.
```

Ogni componente React è una funzione JavaScript che può contenere del markup che React renderizza nel browser. I componenti React utilizzano un'estensione di sintassi chiamata JSX per rappresentare quel markup. JSX assomiglia molto all'HTML, ma è un po' più rigoroso e può visualizzare informazioni dinamiche. La migliore maniera per comprenderlo è quella di convertire un po' di markup HTML in markup JSX.

<Note>

JSX e React sono due cose separate. Sono spesso utilizzati insieme, ma *puoi* [utilizzarli indipendentemente](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform) l'uno dall'altro. JSX è un'estensione di sintassi, mentre React è una libreria JavaScript.

</Note>

> ## Convertire HTML in JSX  {/*converting-html-to-jsx*/}

Reviewer: ```suggestion
## Convertire HTML in JSX {/*converting-html-to-jsx*/}
```

> Supponiamo di avere qualche HTML(perfettamente valido):

Reviewer: ```suggestion
Supponiamo di avere qualche HTML (perfettamente valido):
```

</ul>
```

> E vogliamo inserirlo nel nostro componente:

Reviewer: ```suggestion
E di volerlo inserire nel nostro componente:
```

}
```

> Se lo copiamo e incolliamo così com'è, non funzionerà:

Reviewer: ```suggestion
Copiandolo e incollandolo così com'è, non funzionerà:
```

@Ago95Dev leggendo l'intera frase, comprese le due righe precedenti, sembra suonar meglio, ma onestamente non sono convinto al 100% della soluzione che ti ho proposto 🤔 
Tu cosa ne pensi?

</>
```

> ### 3. Scrivi in camelCase <s>tutte</s> quasi tutte le cose! {/*3-camelcase-salls-most-of-the-things*/}

Reviewer: ```suggestion
### 3. Scrivi in camelCase quasi <s>tutte</s> le cose! {/*3-camelcase-salls-most-of-the-things*/}
```

/>
```

> Puoi [trovare tutti questi attributi nell'elenco delle props del componente DOM.](/reference/react-dom/components/common) Se ne sbagli uno, non preoccuparti: React stamperà un messaggio con una possibile correzione nella [browser console.](https://developer.mozilla.org/docs/Tools/Browser_Console)

Reviewer: ```suggestion
Puoi [trovare tutti questi attributi nell'elenco delle props del componente DOM.](/reference/react-dom/components/common) Se ne sbagli uno, non preoccuparti: React stamperà un messaggio con una possibile correzione nella [console del browser.](https://developer.mozilla.org/docs/Tools/Browser_Console)
```

<Pitfall>

Per ragioni storiche, gli attributi [`aria-*`](https://developer.mozilla.org/docs/Web/Accessibility/ARIA) e [`data-*`](https://developer.mozilla.org/docs/Learn/HTML/Howto/Use_data_attributes) sono scritti come in HTML con i trattini.

</Pitfall>

> ### Pro-tip: Usa un convertitore JSX {/*pro-tip-use-a-jsx-converter*/}

Reviewer: ```suggestion
### Pro-tip: Usa un Convertitore JSX {/*pro-tip-use-a-jsx-converter*/}
```
Come sopra, terrei le maiuscole per convenzione 😄

> Convertire tutti questi attributi nel markup esistente può essere noioso! Raccomandiamo di utilizzare un [converter](https://transform.tools/html-to-jsx) per tradurre il vostro HTML e SVG esistenti in JSX. I convertitori sono molto utili nella pratica, ma è comunque utile capire cosa succede in modo da poter scrivere JSX autonomamente.

Reviewer: ```suggestion
Convertire tutti questi attributi nel markup esistente può essere noioso! Raccomandiamo di utilizzare un [convertitore](https://transform.tools/html-to-jsx) per tradurre il tuo HTML e SVG esistenti in JSX. I convertitori sono molto utili nella pratica, ma è comunque utile capire cosa succede in modo da poter scrivere JSX autonomamente.
```
In alcune sezioni della documentazione diamo del "tu", userei la stessa convenzione qui :)

<Recap>

Ora sai perché JSX esiste e come usarlo nei componenti:

* I componenti React raggruppano la logica di rendering insieme al markup poiché sono correlati.

> * JSX è simile all'HTML, con alcune differenze. Puoi usare un [converter](https://transform.tools/html-to-jsx) se ne hai bisogno.

Reviewer: ```suggestion
* JSX è simile all'HTML, con alcune differenze. Puoi usare un [convertitore](https://transform.tools/html-to-jsx) se ne hai bisogno.
```
